### ISSUE TITLE
Plugin does not take into account time signature?

### ISSUE THREAD

**Issue by DaforLynx** [lemerchand/lemerchand#6]
I have a project using 12/8 time. I'm trying to select every first beat of each measure. But it instead selects each note every fourth beat. This seems to be a rather large oversight. Why doesn't it use the timeline already built into Reaper? I can see each beat being counted. You have UI for beats all the way up to 16. I must be misunderstanding how this plugin works, but it would make sense *to me* for the duration selection to filter by duration, and the beat selection to filter by beat.

Well, after some time wrestling with this I figure it's down to a pretty deep design challenge. The "beat" selection actually refers to "sixteenth notes". This is handy specifically for 4/4 pieces with a minimum used grid of 16th notes. I hope you're able to figure out a way to fix this. I suggest first renaming those functions in the interface to not refer to "beats". Then, have a third selection criteria for the "grid size" used in the beat selection (so single *beat*, half *beat*, third *beat*, and quarter *beat*). So if I select 1/8 duration, beats 1, 4, 7, 10, and a "grid size" of 3, it would internally split each *beat* into thirds, then select every first, fourth, seventh, and tenth "portion" that is of 1/8th note in duration. If the piece is in 12/8, it would split each beat (comprising of three eighth notes) into individual eighth notes, essentially. This should cover most use cases (including mine) and even cover selecting triplet notes that do not fall on the sixteenth note grid - at the cost of, potentially, a bit of confusion from some first-time users.

**Comment by DaforLynx** [comment]
No, I _am_ misunderstanding. A beat is still one eighth note in 12/8 - not 3. So my solution would not work as stated, but it could still work if you significantly expanded the amount of beats able to be selected. But it doesn't change the fact the plugin says "beats" but uses sixteenth notes.

**Comment by lemerchand** [comment]
The README explicitly states that as of right now the beat selector does not work on odd times. I know it would be nice to use it for them, but I am afraid it just wasn't a feature I had the time to add in at the time, and these days I'm too busy and haven't touched Lua in years. Sorry for the inconvenience, but I hope you find the script useful in other scenarios.

**Comment by lemerchand** [comment]
Closing this as it is not something I intend on implementing anytime soon.
